storage/src/backend/registry.rs

```diff
@@ -189,7 +189,14 @@ struct RegistryState {
 fn needs_fallback_http(e: &dyn Error) -> bool {
     match e.source() {
         Some(err) => match err.source() {
-            Some(err) => err.to_string().contains("error:1408F10B"),
+            Some(err) => {
+                let msg = err.to_string().to_lowercase();
+                let fallback = msg.contains("ssl") && msg.contains("record");
```

Comment:
This is a literal string match. I wonder if it works with different registries.

Reply:
It's a simplified client implementation referenced in go: https://github.com/golang/go/issues/44855, it seems for different client using different ssl library, we got different error message:

curl:

```
curl https://localhost:5000/v2
curl: (35) error:0A00010B:SSL routines::wrong version number
```

rust reqwest with current openssl lib:

```
error:1408F10B:SSL routines:ssl3_get_record:wrong version number:ssl/record/ssl3_record.c:331
```

Reply:
@bergwolf Updated, PTAL.